File: frontend/lib/types/today.ts
// Source: backend/app/schemas/today.py

export type TodayFeedItem = {
  account_id: string;
  account_name: string;
  domain?: string | null;
  total_score: number;
  sales_ready: boolean;
  source: "web_scan" | "media_scan";
  conversation_delta?: number | null;
  why_now: string;
  reason_tags: string[];
  updated_at: string;
};

export type TodayFeedResponse = {
  items: TodayFeedItem[];
  total: number;
  generated_at: string;
};

File: backend/app/schemas/today.py
"""Today feed schemas."""

from __future__ import annotations

from pydantic import BaseModel


class TodayFeedItem(BaseModel):
    account_id: str
    account_name: str
    domain: str | None = None
    total_score: int
    sales_ready: bool
    source: str
    conversation_delta: int | None = None
    why_now: str
    reason_tags: list[str]
    updated_at: str


class TodayFeedResponse(BaseModel):
    items: list[TodayFeedItem]
    total: int
    generated_at: str
